driver/task.go

```diff
@@ -44,54 +47,295 @@ type BufferPeriod struct {
+
+	names := make([]string, len(t.services))
+	for ix, s := range t.services {
 		names[ix] = s.Name
 	}
 	return names
 }
 
+// Source returns the module source for the task
+func (t *Task) Source() string {
+	t.mu.RLock()
+	defer t.mu.RUnlock()
+	return t.source
+}
+
+// UserDefinedMeta returns metadata grouped by service
+func (t *Task) UserDefinedMeta() map[string]map[string]string {
+	t.mu.RLock()
+	defer t.mu.RUnlock()
+	return t.userDefinedMeta
+}
+
+// VariableFiles returns a copy of the list of configured variable files
+// for the task's module.
+func (t *Task) VariableFiles() []string {
+	t.mu.RLock()
+	defer t.mu.RUnlock()
+
+	varFiles := make([]string, len(t.varFiles))
+	for i, vf := range t.varFiles {
+		varFiles[i] = vf
+	}
+	return varFiles
+}
+
+// Variables returns a copy of the loaded input variables for a module
+// from configured variable files.
+func (t *Task) Variables() hcltmpl.Variables {
+	t.mu.RLock()
+	defer t.mu.RUnlock()
+
+	vars := make(hcltmpl.Variables)
+	for k, v := range t.variables {
+		vars[k] = v
+	}
+	return vars
+}
+
+// Version returns the configured version for the module of the task
+func (t *Task) Version() string {
+	t.mu.RLock()
+	defer t.mu.RUnlock()
+	return t.version
+}
+
+// configureRootModuleInput sets task values for the module input.
+func (t *Task) configureRootModuleInput(input *tftmpl.RootModuleInputData) {
+	t.mu.RLock()
+	defer t.mu.RUnlock()
+
+	input.Task = tftmpl.Task{
+		Description: t.description,
+		Name:        t.name,
+		Source:      t.source,
+		Version:     t.version,
+	}
+
+	input.Services = make([]tftmpl.Service, len(t.services))
+	for i, s := range t.services {
+		input.Services[i] = tftmpl.Service{
+			Datacenter:  s.Datacenter,
+			Description: s.Description,
+			Name:        s.Name,
+			Namespace:   s.Namespace,
+			Tag:         s.Tag,
+		}
+	}
+
+	var condition tftmpl.Condition
+	switch v := t.condition.(type) {
+	case *config.CatalogServicesConditionConfig:
+		condition = &tftmpl.CatalogServicesCondition{
+			Regexp:            *v.Regexp,
+			SourceIncludesVar: *v.SourceIncludesVar,
+			Datacenter:        *v.Datacenter,
+			Namespace:         *v.Namespace,
+			NodeMeta:          v.NodeMeta,
+		}
+	case *config.ServicesConditionConfig:
+		condition = &tftmpl.ServicesCondition{}
+	default:
+		// expected only for test scenarios
+		log.Printf("[WARN] (driver.terraform) task '%s' condition config unset."+
+			" defaulting to services condition", t.name)
+		condition = &tftmpl.ServicesCondition{}
+	}
+	input.Condition = condition
+
+	input.Providers = t.providers.ProviderBlocks()
+	input.ProviderInfo = make(map[string]interface{})
+	for k, v := range t.providerInfo {
+		input.ProviderInfo[k] = v
+	}
+
+	input.Variables = make(hcltmpl.Variables)
+	for k, v := range t.variables {
+		input.Variables[k] = v
+	}
+}
+
 // clientConfig configures a driver client for a task
 type clientConfig struct {
-	task       Task
 	clientType string
 	log        bool
+	name       string
```

Comment:
optional thought - I could see myself not being sure if `name` meant client name. Would maybe taskName work ok to clarify 'whose' name should be configured here?

Reply:
sure thing, thanks for always looking out for maintainability of code